core/store/src/cold_storage.rs

```diff
@@ -22,6 +22,17 @@ struct StoreWithCache<'a> {
     cache: StoreCache,
 }
 
+/// The BatchTransaction can be used to write multiple set operations to the cold db in batches.
+/// [`write`] is called every time `transaction_size` overgrows `threshold_transaction_size`.
```

Comment:
I think in the previous version you had Drop implemented for this class. What happened to it? Either way I think it would be good to mention that that users needs or doesn't need to call write at the end of processing.